### PR TITLE
Update the 'Projects using Sphinx' page

### PR DESCRIPTION
**Pull request by timhoffm**
Minor updates:

- Add links to documentation of third-party themes
- Add Furo theme and a few examples
- Move a few entries that I happen to know have changed the theme

### CODE REVIEW

File: EXAMPLES.rst [hunk 1]
```diff
@@ -1,10 +1,10 @@
 Projects using Sphinx
 =====================
 
-This is an (incomplete) alphabetic list of projects that use Sphinx or
-are experimenting with using it for their documentation.  If you like to
-be included, please mail to `the Google group
-<https://groups.google.com/forum/#!forum/sphinx-users>`_.
+This is an incomplete list of projects that use Sphinx for their documentation.
+If you like to be included, please create a `pull request on GitHub
```
Comment (jayaddison) [inline EXAMPLES.rst]:
```suggestion
If you would like to be included, please create a `pull request on GitHub
```

(or perhaps, "if you would like a project to be included", or something along those lines)

File: EXAMPLES.rst [hunk 2]
```diff
@@ -311,10 +314,13 @@ Documentation using sphinx_bootstrap_theme
 * `PyUblas <https://documen.tician.de/pyublas/>`__
 * `seaborn <https://seaborn.pydata.org/>`__
 
-Documentation using pydata_sphinx_theme
+Documentation using Pydata Sphinx Theme
```
Comment (jayaddison) [inline EXAMPLES.rst]:
```suggestion
Documentation using PyData Sphinx Theme
```

File: EXAMPLES.rst (hunk 1 above)
Comment (timhoffm) [inline EXAMPLES.rst]:
```suggestion
If you would like to add a project, please create a `pull request on GitHub
```

**Review by jayaddison — APPROVED** [review]

File: EXAMPLES.rst [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 =====================
 
-This is an (incomplete) alphabetic list of projects that use Sphinx or
-are experimenting with using it for their documentation.  If you like to
-be included, please mail to `the Google group
-<https://groups.google.com/forum/#!forum/sphinx-users>`_.
+This is an incomplete list of projects that use Sphinx for their documentation.
+If you would like to add a project, please create a `pull request on GitHub
+<https://github.com/sphinx-doc/sphinx/>`_ or mail to the
+`sphinx-users Google group <https://groups.google.com/forum/#!forum/sphinx-users>`_.
```
Comment (AA-Turner) [inline EXAMPLES.rst]: Perhaps redirect people to a standing GH discussion or issue -- I don't subscribe to 'Sphinx-users' personally.

A

**Review by AA-Turner — APPROVED** [review]

**Pull request merged**